### PR TITLE
Add recipe for r-clang-activation

### PR DESCRIPTION
**Pull request by xhochy**
Also tested that this works locally with the r-arrow-feedstock.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by xhochy** [comment]
Depends on https://github.com/conda-forge/clang-win-activation-feedstock/pull/6

**Comment by xhochy** [comment]
@conda-forge/staged-recipes This is ready for review.

**Comment by xhochy** [comment]
Depends on https://github.com/conda-forge/clang-win-activation-feedstock/pull/8 and a `conda-smithy` release.

File: recipes/r-clang-activation/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  home: https://github.com/conda-forge/r-clang-activation-feedstock
+  license: BSD-3-Clause
+  license_file: LICENSE.txt
+  summary: Activation scripts for using Clang as a compiler in R packages
+  description: |
+    Use clang as an alternative compiler for R packages. Add this as a
+    build requirement using the Jinja function `compiler('r_clang')`.
+
+extra:
+  feedstock-name: r-clang-activation
```
Comment (isuruf) [inline recipes/r-clang-activation/meta.yaml:36]: Can you change this to `r_clang_activation`?

File: recipes/r-clang-activation/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  name: "r_clang_{{ r_clang_target_platform }}"
+  version: {{ version }}
+
+build:
+  number: 0
+  skip: true  # [not win]
+
+requirements:
+  run:
+    - "{{ compiler('clang') }}  {{ version }}"  # [win]
```
Comment (isuruf) [inline recipes/r-clang-activation/meta.yaml:14]: `compiler` macro has a version field, so using the suggestion below is future proof.
```suggestion
    - clang_{{ target_platform }}  {{ version }}
```

**Comment by xhochy** [comment]
Usage of this can be seen in https://github.com/conda-forge/r-arrow-feedstock/pull/16, in contrast to all the sed-hotfixing I have done before, this works pleasantly well.

**Comment by xhochy** [comment]
Fixed CI, still had the `_` in my variables I used in debugging to get rid of `-fno-plt`. Removed this and all is building fine.

**Comment by xhochy** [comment]
@conda-forge/r @isuruf Can I get another look/merge, please?

File: recipes/r-clang-activation/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
+{% set version = "9.0.1" %}
+
+package:
+  name: "r_clang_{{ r_clang_target_platform }}"
```
Comment (isuruf) [inline recipes/r-clang-activation/meta.yaml:4]:
```suggestion
  name: "r_clang_{{ target_platform }}"
```

**Comment by xhochy** [comment]
@isuruf do I then need the `conda_build_config.yml` at all or can I remove it?

**Comment by xhochy** [comment]
> @isuruf do I then need the `conda_build_config.yml` at all or can I remove it?

Answered it myself: It's not need. 

Can you have another look?

**Comment by xhochy** [comment]
@conda-forge/r @conda-forge/r-base Can you take a look/merge? This successfully works together with the `r-arrow` package on Windows. Happy to explain details, I'm eager to get this in the next days.

**Comment by dpryan79** [comment]
I'm not sure how useful Björn or I are when it comes to reviewing stuff like this for windows. If it's making things easier to get r-arrow working then :+1: from me.

**Comment by xhochy** [comment]
If there are no further objections, I'll merge tomorrow morning.

**Comment by bgruening** [comment]
No objections from my side, but I struggel to find any Windows box to test this :(

**Pull request merged**